scripts/generate_api_docs_3.js

```diff
@@ -0,0 +1,501 @@
+    }
+  }
+  (analysis.metadata && analysis.metadata.polymer && analysis.metadata.polymer.behaviors || []).map(f => index(f, 'behaviors'));
+  (analysis.elements || []).map(f => index(f, 'elements'));
+  (analysis.classes || []).map(f => index(f, 'classes'));
+  (analysis.functions || []).map(f => index(f, 'functions'));
+  (analysis.mixins || []).map(f => index(f, 'mixins'));
+  return map;
+}
+
+/**
+ * @param {format.Analysis} analysis
+ * @param {FeatureIndex} index
+ */
+function getIndexPage(analysis, index) {
+  // We filter out the toplevel files because we want to document them first,
+  // and with a bit more detail because they're the main entrypoints.
+  const filenames = [...index.keys()].sort().filter(fn => fn.startsWith('lib'));
+  const withoutLib = filenames.map(fn => {
+    if (fn.startsWith('lib/')) {
+      return fn.slice(4);
+    }
+    throw new Error(`${fn} was expected to be in lib/`);
+  });
+  /** @type {Map<string, string[]>} */
+  const filenamesBySubdir = new Map();
+  for (const filename of filenames) {
+    const subdir = getSubdir(filename);
+    const inSubdir = filenamesBySubdir.get(subdir) || [];
+    inSubdir.push(filename);
+    filenamesBySubdir.set(subdir, inSubdir);
+  }
+
+  return `{% set markdown = "true" %}
+{% set title = "Polymer API Docs" %}
+{% extends "templates/base-devguide.html" %}
+{% block title %} Polymer API Reference {% endblock %}
+{% block content %}
+
+<style>
+  .apidocs-main {
+    padding: 20px 40px;
+  }
+  .apidocs-main section {
+    padding: 20px 0px;
+  }
+  .apidocs-main section>* {
+    margin: 0px;
+  }
+  .apidocs-main a {
+    color: currentColor;
+    text-decoration: none;
+  }
+  .apidocs-main a:hover {
+    color: rgb(21, 101, 192);
+  }
+  .apidocs-main .name a {
+    font-weight: bold;
+  }
+</style>
+<div class="apidocs-main">
+<section>
+  <div class="name">
+    <a href="polymer-element">
+      PolymerElement
+    </a>
+  </div>
+  <marked-element sanitize markdown="Base class that provides the core API for
+    Polymer's meta-programming features including template stamping,
+    data-binding, attribute deserialization, and property change observation.">
+  </marked-element>
+</section>
+
+${[...new Set(['elements', 'mixins', 'utils', 'legacy', ...filenamesBySubdir.keys()])].map((section) => getIndexPageSubsection(section, filenamesBySubdir.get(section), index)).join('\n\n')}
+
+</div>
+{% endblock %}
+`
+}
+
+/** @param {string} fn */
+function getSubdir(fn) {
+  return fn.split('/')[1];
+}
+
+/**
+ *
+ * @param {string} subsection
+ * @param {Array<string>} filenamesIn
+ * @param {FeatureIndex} featureIndex
+ */
+function getIndexPageSubsection(subsection, filenamesIn, featureIndex) {
+  return `
+
+<section anchor-id="${subsection}">
+  <h2>${subsection}</h2>
+
+  ${filenamesIn.map(fn => getFilenameDescription(fn, featureIndex)).join('\n\n')}
+</section>
+`
+}
+
+const hardcodedDescriptions = new Map([
+  ['lib/elements/array-selector.js', `
```

Comment:
What's the way forward here? Can you open issues in Analyzer and Polymer to add these to the source?